### DISCUSSION TITLE
Problem with Pipewire on Linux (but not with pulseaudio)

### DISCUSSION THREAD

**Discussion by trekfan1** [rustdesk/rustdesk#10197] [Q&A]
When I open a new connection, the sounds become distorted but only with pipewire-alsa and pipewire-audio, if I use pulseaudio no problem

**Comment by xlionjuan** [comment]
What distro you're using?

I have this too on my Bluefin (Fedora)41.

**Comment by trekfan1** [comment]
Ubuntu 24.10 Oracular

**Reply by xlionjuan** [reply]
both are new, but I have no idea what will affect, it show up recently

**Comment by 21pages** [comment]
Could you try out https://github.com/21pages/rustdesk/releases/tag/revert_linux_use_cpal_build on the local side?

**Reply by xlionjuan** [reply]
It fixed.

**Reply by jeremiah-k** [reply]
This issue has been affecting me for a few weeks (Linux Mint 22). I tried this release and it's resolved.  Thank you!

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/pull/10326

**Comment by Cabus1982** [comment]
On Linux, regardless of the distro, I always experience slow (relayed) WAN connections, while on Windows, they are mostly fast (direct).

**Reply by xlionjuan** [reply]
I guess the hole punch attempt on Linux is slower?

Maybe you could enable "Always connect via relay" to bypass this.

**Reply by Cabus1982** [reply]
That's what I think, though it doesn't make much sense because we're not talking about an advanced feature. It's an elemental thing for a support tool to be responsive. 'Direct' (LAN) performance is perfect (15 ms), but 'relayed' (WAN) latency reaches up to 600 ms, which makes the tool unusable. 
I've tried tweaking everything, and nothing has helped. It's a shame, especially considering the program has been in development for 4 years. 😯

**Reply by xlionjuan** [reply]
The connection quality is depend on the connection between clients and the relay server, if you're still using public server, **do not blame**, you should build your own server, or with smaller use case, use Tailscale to achieve P2P connections.

**Reply by 21pages** [reply]
Could you try out https://github.com/21pages/rustdesk/releases/tag/qos both sides and see if there is noticeable improvement?

**Reply by Cabus1982** [reply]
> The connection quality is depend on the connection between clients and the relay server, if you're still using public server, **do not blame**, you should build your own server, or with smaller use case, use Tailscale to achieve P2P connections.

It doesn't make sense, because on Windows, connections are mostly fast (Direct), so it's not a matter of not having a server.

**Reply by xlionjuan** [reply]
> but 'relayed' (WAN) latency reaches up to 600 ms

???????

**Reply by Cabus1982** [reply]
Yes, it fluctuates between 300 and 500 ms.

**Reply by xlionjuan** [reply]
So are you using public server?

**Reply by Cabus1982** [reply]
yeap

**Reply by xlionjuan** [reply]
Normal

**Comment by TomEFFENJones** [comment]
I just want to throw my hat into this conversation.

openSUSE Tumbleweed, fully updated. Audio issues when using RustDesk and then trying to play a YouTube video or song through Cider will cause not only audio to not function whatsoever, but also slows the YouTube video down to be choppy, with no audio. I am running on Wayland as xorg runs like garbage on my setup for whatever reason.  I am happy to help any way I can!

I am still a relative noob when it comes to Linux though, so if you do not believe it is an issue with RustDesk, I will try to reach out to the openSUSE community about it. 

Either way, thank you so much for the work you put into this.  It makes managing my devices (Steam Deck especially!) so much easier, all from one device!

**Reply by xlionjuan** [reply]
> openSUSE Tumbleweed, fully updated. Audio issues when using RustDesk and then trying to play a YouTube video or song through Cider will cause not only audio to not function whatsoever

Will browser cause this problem?

**Reply by TomEFFENJones** [reply]
> > openSUSE Tumbleweed, fully updated. Audio issues when using RustDesk and then trying to play a YouTube video or song through Cider will cause not only audio to not function whatsoever
> 
> Will browser cause this problem?

So sorry to take forever to reply!  I want to update you and say that after upgrading to 1.3.6 fixed the audio issue.  To answer your question, yes, it was on Firefox and through the Cider app for Apple Music.  

Awesome work!  Thank you!
